src/framework/piral-core/src/components/ForeignComponentContainer.test.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { ForeignComponentContainer } from './ForeignComponentContainer';
+import { Atom } from '@dbeining/react-atom';
+import { mount } from 'enzyme';
+import { createActions } from '../state';
+import { createListener } from 'piral-base';
+
+const MyComponent = ({ name }) => (
+    <div>{name}</div>
+);
+
+describe('ForeignComponentContainer Module', () => {
+    it('renders correctly for desktop', () => {
+        const state = Atom.of({
+            app: {},
+            rounter: {}
+        });
+        const ctx = createActions(state, createListener({}));
+        const node = mount(<ForeignComponentContainer $portalId="123" $component={MyComponent} $context={ctx} innerProps={name = "Test"} />);
```

Comment:
The $component in here should be a "foreign component", i.e., an object holding a lifecycle based on a (DOM) element.

The `div` is the carrier, so it will always render the expectation in here does not proof anything, unfortunately.